### ISSUE TITLE
Read access is allowed from event dispatch thread or inside read-action only

### ISSUE THREAD

**Issue by ewirch** [xylo/intellij-postfix-templates#33]
```
Read access is allowed from event dispatch thread or inside read-action only (see com.intellij.openapi.application.Application.runReadAction())
Details: Current thread: Thread[ApplicationImpl pooled thread 10,4,main] 1370032623
; dispatch thread: false; isDispatchThread(): false
SystemEventQueueThread: Thread[AWT-EventQueue-0 2017.2.3#IU-172.3968.1 IDEA, eap:true, os:Linux 4.12.7-1-manjaro, java-version:JetBrains s.r.o 1.8.0_152-release-915-b10,6,main] 1441282274
java.lang.Throwable
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:150)
	at com.intellij.openapi.application.impl.ApplicationImpl.assertReadAccessAllowed(ApplicationImpl.java:1075)
	at com.intellij.psi.impl.file.impl.FileManagerImpl.findFile(FileManagerImpl.java:397)
	at com.intellij.psi.impl.PsiManagerImpl.findFile(PsiManagerImpl.java:188)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.loadTemplatesFrom(CustomPostfixTemplateProvider.java:134)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.loadTemplatesFrom(CustomPostfixTemplateProvider.java:117)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.lambda$reloadTemplates$0(CustomPostfixTemplateProvider.java:103)
	at java.util.Optional.ifPresent(Optional.java:159)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.reloadTemplates(CustomPostfixTemplateProvider.java:101)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.reload(CustomPostfixTemplateProvider.java:93)
	at de.endrullis.idea.postfixtemplates.templates.CustomPostfixTemplateProvider.<init>(CustomPostfixTemplateProvider.java:84)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:103)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:80)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:63)
	at com.intellij.openapi.extensions.AbstractExtensionPointBean.instantiate(AbstractExtensionPointBean.java:75)
	at com.intellij.openapi.extensions.CustomLoadingExtensionPointBean.instantiateExtension(CustomLoadingExtensionPointBean.java:47)
	at com.intellij.lang.LanguageExtensionPoint.access$000(LanguageExtensionPoint.java:28)
	at com.intellij.lang.LanguageExtensionPoint$1.compute(LanguageExtensionPoint.java:42)
	at com.intellij.openapi.util.NotNullLazyValue.getValue(NotNullLazyValue.java:39)
	at com.intellij.lang.LanguageExtensionPoint.getInstance(LanguageExtensionPoint.java:53)
	at com.intellij.codeInsight.template.postfix.settings.PostfixTemplatesConfigurable.<init>(PostfixTemplatesConfigurable.java:88)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.picocontainer.defaults.InstantiatingComponentAdapter.newInstance(InstantiatingComponentAdapter.java:193)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.doGetComponentInstance(CachingConstructorInjectionComponentAdapter.java:103)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.instantiateGuarded(CachingConstructorInjectionComponentAdapter.java:80)
	at com.intellij.util.pico.CachingConstructorInjectionComponentAdapter.getComponentInstance(CachingConstructorInjectionComponentAdapter.java:63)
	at com.intellij.openapi.extensions.AbstractExtensionPointBean.instantiate(AbstractExtensionPointBean.java:75)
	at com.intellij.openapi.options.ConfigurableEP$ClassProducer.createElement(ConfigurableEP.java:341)
	at com.intellij.openapi.options.ConfigurableEP.createConfigurable(ConfigurableEP.java:267)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.a(ConfigurableWrapper.java:58)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.getConfigurable(ConfigurableWrapper.java:117)
	at com.intellij.openapi.options.ex.ConfigurableWrapper.cast(ConfigurableWrapper.java:98)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:42)
	at com.intellij.openapi.options.ex.ConfigurableCardPanel.prepare(ConfigurableCardPanel.java:35)
	at com.intellij.ui.CardLayoutPanel.a(CardLayoutPanel.java:128)
	at com.intellij.openapi.application.impl.ApplicationImpl$2.run(ApplicationImpl.java:342)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by xylo** [comment]
Thanks for the bug report.  Do you know how to reproduce the bug?

**Comment by xylo** [comment]
Fixed in version 1.0.2.
